Unicode/makeutype.c

```diff
@@ -630,6 +683,178 @@ static void dumparabicdata(FILE *header) {
     fclose( data );
 }
 
+static void buildtables(FILE *data,long unsigned int *dt,int s,int m,char *t) {
+/* Build ligature/vulgar/fraction table as a condensed uint16 array for the */
+/* range of 't'16{0...s} and then as uint32 for the range 't'32{(s+1)...m}. */
+    int i,j;
+
+    fprintf( data, "const uint16 %s16[] = {\n", t );
+    for ( i=j=0; i<s; i=i+j ) {
+	fprintf( data, "  0x%04x", dt[i] );
+	for ( j=1; j<8 && i+j<s; ++j ) {
+	    fprintf(data, ", 0x%04x", dt[i+j] );
+	}
+	if ( i+j>=s )
+	    fprintf(data, "};\n\n" );
+	else
+	    fprintf(data, ",\n" );
+    }
+    if ( s<m ) {
+	fprintf( data, "const uint32 %s32[] = {\n", t );
+	for ( i=s; i<m; i=i+j ) {
+	    fprintf( data, "  0x%08x", dt[i] );
+	    for ( j=1; j<4 && i+j<m; ++j ) {
+		fprintf(data, ", 0x%08x", dt[i+j] );
+	    }
+	    if ( i+j>=m )
+		fprintf(data, "};\n\n" );
+	    else
+		fprintf(data, ",\n" );
+	}
+    }
+}
+
+static void dumpbsearch(FILE *data,int s) {
+/* Customize bsearch sizes to match data table sizes for uint16 and uint32. */
+    fprintf( data, "static int compare_codepoints%d(const void *uCode1, const void *uCode2) {\n", s );
+    fprintf( data, "    const uint%d *cp1 = (const uint%d *)(uCode1);\n", s, s );
+    fprintf( data, "    const uint%d *cp2 = (const uint%d *)(uCode2);\n", s, s );
+    fprintf( data, "    return( (*cp1 < *cp2) ? -1 : ((*cp1 == *cp2) ? 0 : 1) );\n}\n\n" );
+}
+
+static void dump_getU(FILE *data,long unsigned int *dt,int s,int m,char *t, char *n) {
+/* Get the 'n'th unicode value in the table, where {0<=n<=(sizeof(table)-1} */
+/* Treat both (uint16, uint32) tables as if it is only one table. Error=-1. */
+    fprintf( data, "int32 %s_get_U(int n) {\n", n );
+    fprintf( data, "    if ( n<0 || n>=%d )\n\treturn( -1 );\n    ", m );
+    if ( s<m )
+	fprintf( data, "if ( n<%d )\n\t", s );
+    fprintf( data, "return( (int32)(%s16[n]) );\n", t );
+    if ( s<m )
+    fprintf( data, "    else\n\treturn( (int32)(%s32[n-%d]) );\n", t, s );
+    fprintf( data, "}\n\n" );
+}
```

Comment:
'm' is the entire maximum length of the table *dt which will got from reading unicode list and then be named 't' when building the tables and functions in is_Ligatures.c. Looking at a byte count, it made less waste to have an array of 500+ ligatures, 20+ vulgars, and about 50 odd fraction definitions (approx 600 x 4bytes = 2400bytes) instead of assigning an individual 1bit for each in the lookup table (approx 1bit x MAXC = 1/8byte x 65536 = 8k). doing this for ligs, vulgs and frac would make it 8k x 3 = 24k used in flag space. Since I was already building a program, it seemed worth going the extra step and compressing everything in {0..65535} into uint16 (so we split the table 0...s), and anything left afterwards goes into {65536....->) which is uint32 (and is the remainder of the table from s+1...m. so instead of using 2400bytes, now is closer to 1200bytes to define about 600 ligs/vuls/fracs.
'n' is used to name the various functions using the tables.
These routines are reusable 3x to build ligs, vuls, fracs., so testing against one verifies the other two will build similarly the same.